packages/ovault-evm/contracts/OVaultComposer.sol

```diff
@@ -149,93 +156,110 @@ contract OVaultComposer is IOVaultComposer, ReentrancyGuard {
-
-        SendParam memory refundSendParam = failedMessage.refundSendParam;
-        address refundOft = failedMessage.refundOFT;
-
-        /// @dev If the destination is the HUB chain, we just transfer the tokens to the receiver
-        if (refundSendParam.dstEid == HUB_EID) {
-            _executeHubTransfer(refundOft, refundSendParam.to.bytes32ToAddress(), refundSendParam.amountLD);
-        } else {
-            refundSendParam.extraOptions = _extraOptions;
-            _send(refundOft, refundSendParam);
+        if ((f != FailedState.CanOnlyRefund) && (f != FailedState.CanRefundOrRetryWithSwap)) revert CanNotRefund(_guid);
+
+        try
+            this.sendFailedMessage{ value: msg.value }(
+                _guid,
+                failedMessage.refundOFT,
+                failedMessage.refundSendParam,
+                failedMessage.msgValue,
+                REFUND_OVERPAY_ADDRESS
+            )
+        {
+            emit Refunded(_guid, failedMessage.refundOFT);
+        } catch {
+            emit SendFailed(_guid, failedMessage.refundOFT);
         }
-
-        emit Refunded(_guid, refundOft);
     }
 
-    /// @dev Permissionless function to retry the message with more gas
-    /// @dev Failure case when there is a LayerZero config issue when a peer is set - ex: dvn config
-    function retry(bytes32 _guid, bytes calldata _extraOptions) external payable nonReentrant {
+    /// @dev If removeExtraOptions is true, only enforcedOptions are used and the sender needs to pay the entire msg.value
+    /// @dev A transaction can never fail due to a lack of extraOptions, unless the OFT is customized in-which case the Composer would need to be customized.
+    function retry(bytes32 _guid, bool removeExtraOptions) external payable nonReentrant {
         FailedMessage memory failedMessage = failedMessages[_guid];
         if (_failedGuidState(failedMessage) != FailedState.CanOnlyRetry) revert CanNotRetry(_guid);
 
-        delete failedMessages[_guid];
-
         SendParam memory sendParam = failedMessage.sendParam;
-        address retryOFT = failedMessage.oft;
+        uint256 prePaidMsgValue = failedMessage.msgValue;
 
-        /// @dev If the destination is the HUB chain, we just transfer the tokens to the receiver
-        if (sendParam.dstEid == HUB_EID) {
-            _executeHubTransfer(retryOFT, sendParam.to.bytes32ToAddress(), sendParam.amountLD);
-        } else {
-            sendParam.extraOptions = _extraOptions;
-            _send(retryOFT, sendParam);
+        if (removeExtraOptions) {
+            sendParam.extraOptions = "";
+            prePaidMsgValue = 0;
+
+            payable(REFUND_OVERPAY_ADDRESS).transfer(prePaidMsgValue);
         }
 
-        emit Retried(_guid, retryOFT);
+        try
+            this.sendFailedMessage{ value: msg.value }(_guid, failedMessage.oft, sendParam, prePaidMsgValue, tx.origin)
+        {
+            emit Retried(_guid, failedMessage.oft);
+        } catch {
```

Comment:
should update the msg.value calculation, also taking into account the amount send to REFUND_OVERPAY_ADDRESS
```solidity
failedMessages[_guid].msgValue = prePaidMsgValue + msg.value;
```